src/server/channel_store.cc

```diff
@@ -23,6 +23,30 @@ bool Matches(string_view pattern, string_view channel) {
   return stringmatchlen(pattern.data(), pattern.size(), channel.data(), channel.size(), 0) == 1;
 }
 
+// Build functor for sending messages to connection
+auto BuildSender(string_view channel, facade::ArgRange messages) {
+  size_t messages_size = accumulate(messages.begin(), messages.end(), 0,
+                                    [](int sum, string_view str) { return sum + str.size(); });
+  auto buf = shared_ptr<char[]>{new char[channel.size() + messages_size]};
+  {
+    memcpy(buf.get(), channel.data(), channel.size());
+    char* ptr = buf.get() + channel.size();
+    for (string_view message : messages) {
+      memcpy(ptr, message.data(), message.size());
+      ptr += message.size();
+    }
+  }
```

Comment:
Why not just store two `std::strings` ? One for `channel` and one for `messages` ? That way we can get rid of the `shared_ptr` which I am not quite sure I understand why it's needed: 

```
>> 442 void Connection::DispatchOperations::operator()(const PubMessage& pub_msg) {
   443   RedisReplyBuilder* rbuilder = (RedisReplyBuilder*)builder;
   444   unsigned i = 0;
   445   array<string_view, 4> arr;
   446   if (pub_msg.pattern.empty()) {
   447     arr[i++] = "message";
   448   } else {
   449     arr[i++] = "pmessage";
   450     arr[i++] = pub_msg.pattern;
   451   }
   452   arr[i++] = pub_msg.Channel();
   453   arr[i++] = pub_msg.Message();
   454   rbuilder->SendStringArr(absl::Span<string_view>{arr.data(), i},
   455                           RedisReplyBuilder::CollectionType::PUSH);
   456 }

```

Reply:
and that way maybe you can even get rid of the functor

Reply:
It was just an over-optimization to reduce memory waste with enqueued messages, because they can take up lots of space